### PR TITLE
Hook up API routes to Postgres

### PR DESCRIPTION
**Pull request by tf3**
Completes the following:
- https://trello.com/c/VvrHgZtW/10-implement-api-routes-for-chosen-dmbs-postgres
- https://trello.com/c/BUyVYWdo/7-implement-tests
- https://trello.com/c/Rl2mshoe/14-clean-up-inherited-seeding-code

### CODE REVIEW

File: server/db-pg/model.js [hunk 1]
```diff
@@ -19,8 +19,8 @@ const parseRestaurant = (restaurant) => {
 
 // Same as `pasreRestaurant`, except it stringifies the relevant values.
```
Comment (garrowat) [inline server/db-pg/model.js:20]: nitpick: spelling of parse

File: server/db-pg/model.js [hunk 2]
```diff
@@ -52,13 +52,14 @@ const updateRestaurant = (id, valuesToUpdate) => {
...
+  return client.query(queryString)
+    .then(res => parseRestaurant(res.rows[0]));
 };
 
 const postRestaurant = (restaurant) => {
   stringifyRestaurant(restaurant);
   const columns = Object.keys(restaurant);
-  const values = Object.values(restaurant); // must these be in single quotes?
+  const values = Object.values(restaurant).map(value => `'${value}'`);
   const queryString = `insert into restaurants(${columns.join(', ')}) values(${values.join(', ')})`;
```
Comment (garrowat) [inline server/db-pg/model.js:63]: Probably not a big deal for our purposes, but I've read that building queries with string concatenation (and I assume string templates) can open you up to injection attacks.

See here: 
https://node-postgres.com/features/queries
Relevant line is  under **Parameterized query**.

File: server/db-pg/seedHelpers.js [hunk 3]
```diff
@@ -8,6 +8,11 @@ const database = process.env.DATABASE_NAME;
 const startId = Number(process.env.START_ID) || 1;
 const endId = Number(process.env.END_ID) || 1000;
 
+// The number of chunks that the restaurant data will be divided into when
+// creating the CSV. Increase this number if you get out-of-memory errors while
+// seeding.
+const numberOfChunks = Number(process.env.NUMBER_OF_CHUNKS) || 100;
```
Comment (garrowat) [inline server/db-pg/seedHelpers.js:14]: Nice user-friendliness here 💯

File: server/db-pg/seedHelpers.js [hunk 4]
```diff
@@ -19,50 +24,27 @@ const makeName = () => lorem.generateWords(2);
...
-const makeAddress = () => Math.floor(Math.random() * Math.floor(100)) + ' ' + lorem.generateWords(1) + ' ' + ['St.', 'Blvd.', 'Rd.'][Math.floor(Math.random() * Math.floor(3))];
+const randomInt = (min, max) => min + Math.floor(Math.random() * max);
+
+const makeAddress = () => `${randomInt(1, 300)} ${lorem.generateWords(1)} ${['St.', 'Blvd.', 'Rd.'][randomInt(0, 2)]}`;
 
-const makeDeliveryTime = () => Math.floor(Math.random() * Math.floor(60));
+const makeDeliveryTime = () => randomInt(1, 60);
 
 // TODO: fix this
 const makeLocation = () => [33.067149, -117.263955];
```
Comment (garrowat) [inline server/db-pg/seedHelpers.js:34]: You may already have plans for this, but I know that Faker can generate fake coords.

File: server/db-pg/seedHelpers.js [hunk 5]
```diff
@@ -19,50 +24,27 @@ const makeName = () => lorem.generateWords(2);
...
-  close: '1100',
-},
-{
-  day: 'Sunday',
-  open: '0500',
-  close: '1100',
-}];
+});
+
+const makeHours = () => {
```
Comment (garrowat) [inline server/db-pg/seedHelpers.js:42]: I like how modular and clear your code is.

File: server/db-pg/model.js (hunk 2 above)
Comment (tf3) [reply server/db-pg/model.js:63]: Thanks, this is good to know! I should definitely fix this before merging.

File: server/index.js [hunk 6]
```diff
@@ -13,44 +13,41 @@ app.use(express.json());
...
-  let deletedRest;
 
-  models.getRest(id)
-    .then(dbResponse => deletedRest = dbResponse)
-    .then(() => models.deleteRest(id))
-    .then(() => res.json(deletedRest))
+  model.deleteRestaurant(id)
+    .then(deletedRestaurant => res.json(deletedRestaurant))
     .catch(err => res.status(202).send(err).end());
 });
```
Comment (garrowat) [inline server/index.js:53]: This looks good to me.

**Review by garrowat — APPROVED** [review]
Looks great. I'm going to go ahead and merge since I don't see anything to fix really.

File: server/db-pg/model.js (hunk 2 above)
Comment (garrowat) [reply server/db-pg/model.js:63]: My bad! I'll make sure to check with you in the future.

Comment (glkapitan) [reply server/db-pg/model.js:63]: interesting.  Garret, what are the options? For example query string like below where text is '"'
How should it be built?
'LOAD DATA LOCAL INFILE ? INTO TABLE tableName  ' ENCLOSED BY ' +  `'${text}'`

**Review by glkapitan — COMMENTED** [review]
Looks really good

**Pull request merged**